### ISSUE TITLE
[patent policy] Exclusion opportunity when work leaves a Working Group

### ISSUE THREAD

**Issue by hober** [w3c/w3process#518]
Under the current process & patent policy, if a Member or Invited Expert leaves a Working Group, the "departing participants have a 60-day Exclusion Opportunity after their actual resignation to exclude Essential Claims." ([Process § 3.6. Resignation from a Group](https://www.w3.org/2020/Process-20200915/#resignation), [Patent Policy § 4.2. Exclusion and Resignation From the Working Group](https://www.w3.org/Consortium/Patent-Policy-20200915/#sec-exclusion-resign))

So this covers the case where an **organization** leaves a group, but AFAICT there isn't an analogous 60-day Exclusion Opportunity given to organizations when **work** leaves a group (e.g., by a deliverable being removed from a charter).

**Comment by frivoal** [comment]
I believe you're right. This case should be quite easily to work around by getting into the habit of publishing a Working Draft prior to dechartering a deliverable, but that's a workaround (which needs the collaboration of WG members), not a fix.

I think it would be worth doing another round of edits to the Patent Policy to try better deal with the fact that groups can open and close, charters can change, and documents can switch groups. Very little of that is addressed explicitly at the moment. I would very much welcome tightening up things, but <abbr title="I am not a lawyer">IANAL</abbr>, and I suspect actual lawyers might not be too eager to tinker with the existing wording, even if vague, so long as it hasn't caused actual issues in practice.

**Comment by dwsinger** [comment]
It is, alas, another example where the mental model that a WG exists to make one spec in one charter period is catching us out again.

**Comment by chaals** [comment]
If work leaves a group, then are there surviving commitments? If a different group publishes a Recommendation, then isn't the grant dependent on being a member of that second group at some time?

(I still think that the underlying problem @dwsinger noted should be fixed, because I **think** the problem here isn't a lack of exclusion opportunity, but the lack of commitments for contributions...)

**Comment by dwsinger** [comment]
roughly, yes. someone I won't name said that "could have excluded and did not" is a useful test, so the lack of an exclusion opp concerns me

**Comment by fantasai** [comment]
This is a general problem around rechartering prior to REC/PRD. You're referring to the expiration of a group's charter, right? Not premature amendment to remove the spec from an existing charter?

PSIG didn't have the appetite to take it up last year, and I suspect they will be uncomfortable making another revision to the Patent Policy so soon after that one. But there's a possibility we can fix it by revision of the Process, so maybe we can look into that for Process 2022. It's been hard to get traction on it, but if Apple thinks this is an issue worth solving, then we have a chance. :) Most recent prior art is [in this message](https://lists.w3.org/Archives/Member/member-psig/2020AprJun/0043.html).

**Comment by dwsinger** [comment]
I think that before we do any transfer of a spec. (especially if we transfer to another _kind_ of venue, such as a CG or the WHATWG), and before we 'park' a spec. (when a WG decides a spec. won't be making progress), we should trigger an exclusion opportunity somehow. Is PRD the right vehicle? Should this be in the Patent Policy (hard), Process (less hard) or Practice (easiest)?

**Comment by fantasai** [comment]
Before "parking" a spec, you of course should publish the latest version on /TR. Nothing outside of /TR is considered for the Patent Policy. (Another reason to publish *frequently* on /TR rather than pretending an ED is a real spec.)

If you snapshot the work as a WD, then W3C can pick up the work and take it to REC later. But as you mention, the commitments on material that was not covered by the FPWD exclusion period are a bit uncertain, as it is not clear if there is an exclusion opportunity for companies that leave the WG by expiration of the charter. Fixing this is a longstanding issue, and will require modifications to either the Patent Policy or the Process.

If you snapshot the work as a 'PRD' (CR or REC) then you trigger all prior commitments to the spec as well as new ones against the current material. This requires no changes to the Patent Policy or the Process, but depending on the state of the draft, might require leaning on the Director to be a bit less rigorous about the transition requirements. If you're planning to transition work out of W3C, I would recommend this route.

**Comment by plehegar** [comment]
(waiting on input from PSIG)

**Comment by wseltzer** [comment]
There's consensus in PSIG that PRD is an appropriate means of gathering patent commitments and calls for exclusion. [minutes](https://www.w3.org/2022/02/14-psig-minutes.html#ResolutionSummary)

PSIG discussed various opinions on whether the Process CG should address whether/how PRDs are initiated when work leaves, without reaching conclusion on this second point.

**Comment by plehegar** [comment]
So the question becomes: should we reduce CR criteria ([6.3.3](https://www.w3.org/2021/Process-20211102/#transition-reqs)/[6.3.7](https://www.w3.org/2021/Process-20211102/#transition-cr) and [6.3.4](https://www.w3.org/2021/Process-20211102/#update-requests)) if the work leaves in order to publish a PRD.
In particular:
- must [formally address](https://www.w3.org/2021/Process-20211102/#formally-addressed) all issues raised about the document since the previous [maturity level](https://www.w3.org/2021/Process-20211102/#maturity-levels).
-  must document how [adequate implementation experience](https://www.w3.org/2021/Process-20211102/#adequate-implementation) will be demonstrated,
- must specify the deadline for comments, which must be at least 28 days after publication, and should be longer for complex documents,
- must show that the changes have received [wide review](https://www.w3.org/2021/Process-20211102/#dfn-wide-review).

Imho, we could allow waiving those requirements if, and only if, the document is republished as a Discontinued Draft after the end of the Call for Exclusion. This prevents those documents from lingering as CR or CRS in /TR.

**Comment by frivoal** [comment]
I am unsure that PSIG got to the bottom of this issue. A PRD would do two things:
* trigger an exclusion opportunity, which binds people who don't exclude to eventually provide a license under W3C Royalty-Free (RF) Licensing Requirements
* makes the commitment to license active at the end of that exclusion opportunity

Normally, exclusion opportunities that bind people to eventually provide a license happen in multiple occasions in addition to publishing a PRD (joining the group, leaving the group, publishing a FPWD). But we deliberately wait until CR (or REC) to call something a PRD, so that the commitments only kick in after sufficient consensus and expectations of interoperability have been established (CR or later).

Forcing a PRD regardless of maturity when works leaves a group would make the commitment to license active even when consensus has not been reached. Personally, I am comfortable with that, but based on previous discussions in PSIG, I think it is not obvious that this would be acceptable to everyone. This may have been considered by PSIG and deemed acceptable, but to my reading, the minutes fall short of making that explicit.

Maybe @wseltzer can offer some insights on whether that was adequately covered in the meeting. If not, it might be worth going back to it.

A related consideration is that unless we add a requirement to do the same at the end of a charter's normal duration, this would mean that works that gets explicitly removed from the scope or from the list of deliverables prior to reaching CR would get stronger patent protection than works that is simply neglected until the charter runs out.

Assuming we confirm that making the license commitment kick in is indeed acceptable, I wouldn't go the route suggested by PLH and forcibly make the last draft a CR with relaxed criteria before discontinuing it. The fact that the patent policy uses the term PRD rather than CR directly does offer us a level of intirection, which means we can deem other things than CRs and RECs to be PRDs. So we could simply say that the last publication is a PRD, without having to make it a CR.

Something like:
> When the charter of a Working Group is amended to reduce its scope or to remove some Recommendation Track deliverables previously included, the Working Group must, before the updated charter comes into effect, republish those technical reports that are affected and that have been published at least once, at least as either a Working Draft or Discontinued Draft as appropriate, or at a higher maturity level if the relevant criteria are met. Regardless of this maturity level, this last publication being de-listing or before scope reduction is deemed a Patent Review Draft for the purpose of the Patent Policy, including the content of candidate changes, if any.

If we want to avoid busy-work republishing things, and if we assume than an already published technical report can retrospectively be made into a PRD, we could go with:

> When the charter of a Working Group is amended to reduce its scope or to remove some Recommendation Track deliverables previously included, the Working Group <del>must</del><ins>should</ins>, before the updated charter comes into effect, republish those technical reports that are affected and that have been published at least once<ins> and for which the Working Group has prepared susbtantive changes that have not yet been published</ins>, at least as either a Working Draft or Discontinued Draft as appropriate, or at a higher maturity level if the relevant criteria are met. Regardless of this maturity level, this last publication before being de-listing or before scope reduction is deemed a Patent Review Draft for the purpose of the Patent Policy, including the content of candidate changes, if any. <ins>For those technical reports affected by the scope reduction or being removed from deliverables that the Working Group chooses not to republish, then they are deemed to become Patent Review Drafts regardless of maturity level if they weren't already, including the content of candidate changes if any, with an effective publication date on the day before the recharter is effective, causing the start of a Exclusion Opportunity.</ins>

The extra clause for group end-of-life could be:
> In addition, when a Working Group charter reaches the end of its duration without being extended, the provision above is applied to all Recommendation Track deliverables of the working group that have been published at least once.

Alternatively, if we merely want to gather licensing commitments without activating them until an actual PRD is reached (in an other group later charted to carry on the work), we should tweak the text above to directly cause an <a href="https://www.w3.org/Consortium/Patent-Policy-20200915/#exclusion-opportunity">Exclusion Opportunity</a> to start, without making the document a PRD. There's plenty of precedent for doing that, since the Patent Policy itself does start exclusion opportunities on number of occasions other than PRD publication. However, unlike PRD which are explicitly meant to be defined in the Process Document, the Patent Policy does not have an explicit statement that the Process may describe additional situations which trigger an Exclusion opportunity, and that might be worth adding.

**Comment by wseltzer** [comment]
As PSIG discussed the issue, it was in the context of an existing PRD publication that involves a call for consensus.

**Comment by wseltzer** [comment]
@frivoal We envisioned actively "publishing" a document as PRD. Under existing process, that involves getting consensus of the WG to publish. PSIG did not agree that a document without such consensus should be blessed as PRD.

**Comment by plehegar** [comment]
Please note that getting a PRD means in practice publishing a Candidate Recommendation Snapshot, which comes additional requirements, such as going through horizontal review. In addition, publishing a document as a "Candidate Recommendation" when there is no intent to make a W3C standard seems awkward. "Discontinued Draft" does not constitute a PRD. Should we allow a Group to request a Call for Exclusion on a Discontinued Draft?

**Comment by plehegar** [comment]
from https://www.w3.org/2023/01/11-w3process-minutes.html
-> deferred

**Comment by dwsinger** [comment]
Work can leave a working group because it's discontinued, or transferred to another WG (and maybe other ways). I am unconvinced that requiring the WG that it's leaving to do a lot of work will result in any of that work getting done.
